bcipy/display/paradigm/vep/display.py

```diff
@@ -0,0 +1,491 @@
+        assert len(code) >= self.refresh_rate, 'Code must be longer than refresh rate'
+        starting_position = position
+        positions = [
+            starting_position,  # bottom left
+            [starting_position[0] + size, starting_position[1]],  # bottom right
+            [starting_position[0] + size, starting_position[1] + size],  # top right
+            [starting_position[0], starting_position[1] + size],  # top left
+        ]
+        box = []
+        for pos in positions:
+            pattern1 = visual.GratingStim(win=self.window, name=f'2x2-1-{pos}', units='height',
+                                          tex=None, mask=None,
+                                          ori=0, pos=pos, size=size, sf=1, phase=0.0,
+                                          color=color[0], colorSpace='rgb', opacity=1,
+                                          texRes=256, interpolate=True, depth=-1.0)
+            pattern2 = visual.GratingStim(win=self.window, name=f'2x2-2-{pos}', units='height',
+                                          tex=None, mask=None,
+                                          ori=0, pos=pos, size=size, sf=1, phase=0,
+                                          color=color[1], colorSpace='rgb', opacity=1,
+                                          texRes=256, interpolate=True, depth=-2.0)
+            color = (color[1], color[0])
+            box += [VEPBox(flicker=[pattern1, pattern2], code=code)]
+
+        self.vep += box
+        return box
+
+    def _trigger_pulse(self) -> None:
+        """Trigger Pulse.
+
+        This method uses a calibration trigger to determine any functional
+            offsets needed for operation with this display. By setting the first_stim_time and searching for the
+            same stimuli output to the marker stream, the offsets between these proceses can be reconciled at the
+            beginning of an experiment. If drift is detected in your experiment, more frequent pulses and offset
+            correction may be required.
+        """
+        calibration_time = _calibration_trigger(
+            self.experiment_clock,
+            trigger_type=self.trigger_type,
+            display=self.window)
+
+        # set the first stim time if not present and first_run to False
+        if not self.first_stim_time:
+            self.first_stim_time = calibration_time[-1]
+            self.first_run = False
+
+    def schedule_to(self, stimuli: List[List[str]], timing: List[List[float]], colors: List[List[str]]) -> None:
+        """Schedule stimuli elements (works as a buffer).
+        """
+        self.stimuli_inquiry = stimuli
+        self.stimuli_timing = timing
+        self.stimuli_colors = colors
+
+    def _build_text_boxes(self) -> List[visual.TextBox2]:
+        """Build the text boxes for the experiment.
+
+        This assumes that vep_type is 4 using a 2x2 grid. Additional configurations can be added in the future.
+        """
+        # generate the inquiry stimuli
```

Comment:
Is the idea that we will have a fixed number of configurations, or did you have thoughts on making this more general and be able to accommodate any possible grid?

Reply:
I really only considered having a few fixed layouts with different size flickering VEP boxes. I'd be open to something more configurable within a fixed unit type (height or norm). We could try looping in code generated via psychopy to give more options. When coding this, I realized we aren't really equipped across the board to support interface building for non-coding folks (only configuration). 

If a user wanted something more custom, I would recommend they build it in the psychopy experiment builder GUI and import bcipy to get the VEP flickering and/or acquisition.